### PR TITLE
Fix alignment for commits on dashboard

### PR DESCRIPTION
**Pull request by mrsdizzie**
Fix allignment issue between commit SHA and commit message

Old:
Screen Shot 2020-05-24 at 1 01 44 AM

New:
Screen Shot 2020-05-25 at 10 08 22 PM

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Hmm this seems to make the text jump for me here:

Screenshot from 2020-05-24 08-49-00

**Review by zeripath — CHANGES_REQUESTED** [review]
doesn't appear to work

**Comment by mrsdizzie** [comment]
@zeripath hmm weird -- does the dashboard on gitea.com look ok to you? Right now for me everything (which is only mirror synching) looks broken like my first screenshot. This change makes them all look normal for me in every browser.

Do you see them as broken but this change doesn't fix it?

**Comment by zeripath** [comment]
No the gitea.com dashboard looks broken too. 

Fiddling a little the issue might be that the SHA link isn't display: inline-block

**Comment by zeripath** [comment]
Sorry or rather that the text.truncate is display: inline-block

**Comment by CirnoT** [comment]
Does not work here either, it's broken with and without this PR, just in different direction. Changing it to -4px works for me but that's a rather poor hack that most likely depend on font too.

**Comment by mrsdizzie** [comment]
Weird, must be some combination of font/os that gives different results it looks totally fine in all 3 browsers on MacOs with this PR but I just tried in a windows vm and see the same thing as @zeripath . 

The fact that its a negative margin makes me feel like its making up for something else that needs to be fixed, though unsetting display: inline-block makes it look mostly better, and causes margin top and bottom to just have no effect. WIll need to look into more

**Comment by silverwind** [comment]
Remove all hacky margins on `img` and `span` and set `<li>` to `display: flex; align-items: center`:

image

The flexbox makes the whitespace collapse so you'll need something like `li > * + * {margin-left: .35rem}` to restore it:

image

**Comment by silverwind** [comment]
Also, I think you can add small top/bottom margins to `<li>` so multiple messages are not stuffed together so tightly, e.g. `.125rem` on both sides.

**Comment by mrsdizzie** [comment]
@silverwind thanks, applied those changes and seems to work better in all places now

**Comment by zeripath** [comment]
Screenshot from 2020-05-26 20-17-43

**Comment by zeripath** [comment]
seems not unfortunately

**Comment by CirnoT** [comment]
Works for me @zeripath

**Comment by silverwind** [comment]
Works for me too:

image

**Review by silverwind — APPROVED** [review]

**Comment by mrsdizzie** [comment]
@zeripath you use Ubuntu right? I set up a vm and think this is an issue with Liberation Mono font which was the font it chose with default settings

It seems to just size much different than others. Removing it from the mono font stack fixes it, but obviously a problem if that is going to be the default font selected for many users. Not really sure what would be a good way of working with that:


May-26-2020 20-18-49

**Comment by silverwind** [comment]
If Liberation Mono is causing issues, maybe we want to move it after all other monospace fonts, right before `monospace`. Thought I'm not sure if any better suitable fonts are present by default on Linux systems.

**Comment by mrsdizzie** [comment]
In this test for Ubuntu it falls back to 'monospace' when removing Liberation Mono which seems to then use DejaVu Sans Mono from the system which looks better imo. Not sure what other distributions have installed other than Liberation Mono but I can't imagine there is only one monospace font installed.

**Comment by silverwind** [comment]
Adding DejaVu Sans Mono before Liberation Mono sounds like a good idea, if it works better.

File: web_src/less/_dashboard.less [hunk 1]
```diff
@@ -80,6 +80,16 @@
 &.feeds {
     .news {
 
+        li {
+            display: flex;
+            align-items: center;
```
Comment (CirnoT) [inline web_src/less/_dashboard.less]:
```suggestion
            align-items: baseline;
```

**Comment by mrsdizzie** [comment]
Google suggest there are several other mono space fonts included in various linux distros -- Id vote to remove liberation mono as a specifically named font all together if it somehow has the ability to look different enough to break styling in some situations compared to all other monospace font.

I'm not a desktop Linux user though so not sure what the real experience with others is. Maybe somebody who is could say.

File: web_src/less/_dashboard.less [hunk 2]
```diff
@@ -80,6 +80,16 @@
...
+            margin-top: .5rem;
+            margin-bottom: .5rem;
+        }
```
Comment (CirnoT) [inline web_src/less/_dashboard.less:92]:
```suggestion
            img {
                align-self: flex-end;
            }
        }
```

**Review by CirnoT — COMMENTED** [review]
How about this?

chrome_2020-05-27_02-54-01

**Comment by mrsdizzie** [comment]
@CirnoT those changes seem to make it look ok and similar enough with Libertion Mono so seem good to me as long as this works for everybody and doesn't look broken as it does now

**Comment by silverwind** [comment]
That `baseline` change does shift all the hash and message 1px up for me, I think I prefer the `center` and fix the fonts.

**Comment by CirnoT** [comment]
Fix the fonts is not a proper solution; you can try manipulate list to look fine to you but in the end you can never be sure what user has installed and what browser will use. Aligning to baseline seems like the only option where you can be sure that different fonts will be aligned to each other visually.

**Review by CirnoT — APPROVED** [review]

**Comment by silverwind** [comment]
I'd say the behaviour in https://github.com/go-gitea/gitea/pull/11595#issuecomment-634224764 is a font/browser bug that should not require is to make specific workarounds for as these may bite us in the long run (after the font/browser is fixed).

**Comment by CirnoT** [comment]
Yes on Libertion Mono the text is around 1px too high compared to avatar, but that's because the font's baseline is off. My proposed changes do not workaround the font issue, they align items to their baseline. This looks off on Libertion Mono for same reason the original solution did, but at least it's less visible now. Other fonts look fine to me and text is aligned perfectly to avatar.

**Comment by CirnoT** [comment]
In the end, baseline is the only thing that is guaranteed to visually align two different font types and I consider it the proper solution long-term.

**Comment by silverwind** [comment]
The alignment looks ok in those screenshots. What I don't quite get is how `align-self: flex-end;` on the `` fixes the image. Care to explain?

**Comment by CirnoT** [comment]
Good question, I have no idea why it actually visually centers image.

File: web_src/less/_dashboard.less [hunk 3]
```diff
@@ -80,6 +80,20 @@
...
     .news {
 
+        li {
+            display: flex;
+            align-items: baseline;
+            margin-top: .5rem;
+            margin-bottom: .5rem;
+
+            img {
+                align-self: flex-end;
```
Comment (CirnoT) [inline web_src/less/_dashboard.less]:
```suggestion
                align-self: flex-start;
```

**Review by CirnoT — COMMENTED** [review]
Try with this and

```css
        .push.news .content ul {
            line-height: 18px;
```

**Comment by CirnoT** [comment]
There does indeed seem to be some misalignment when zoomed out
chrome_2020-05-27_12-09-37

The new proposed changes (flex-start and line-height) seem to eliminate it at the cost of putting the image slightly down when zoomed in; same as on GitHub

**Comment by zeripath** [comment]
it's possibly font x-height differences between the monospace font and the sans-serif font used for the message. 

The solution might simply be to stick the sha in side a rounded-box like we do elsewhere then the jarring font change would be less visible.

**Comment by CirnoT** [comment]
The new changes seem to work rather well for me

Default font on Windows 10:
chrome_2020-05-27_12-16-53
chrome_2020-05-27_12-17-14

Liberation Mono:
chrome_2020-05-27_12-17-43
chrome_2020-05-27_12-17-32

---
For reference, GitHub:
chrome_2020-05-27_12-18-26
chrome_2020-05-27_12-18-41

**Comment by mrsdizzie** [comment]
@CirnoT applied those changes, looks ok to me on various systems/fonts

@zeripath I think one of my least favorite parts of Gitea UI is the SHA1 inside of a rounded box : (  I'd rather see those all eventually get replaced rather than add another one here. These changes seem to make it work and look as it is intended to look now

**Review by CirnoT — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by mrsdizzie** [comment]
I think this should be backported to 1.12 -- I'm not sure exactly where but the bug this fixes seems like it was introduced somewhere in 1.12 (probably one of the larger timeline UI changes). I don't see it on 1.11

**Comment by zeripath** [comment]
Please send backport

**Pull request merged**